### ISSUE TITLE
Implement implicit conversion between F# fun to .NET generic Func<> for ASP.NET minimal API.

### ISSUE THREAD

**Issue by oleksandr-bilyk** [fsharp/fslang-suggestions#1334]
I propose F# implicitly cast F# fun with parameter attributes to C# delegate. It will help to write ASP.NET minimal API in short way

Now C# looks shorter and 
<img width="1177" alt="image" src="https://github.com/fsharp/fslang-suggestions/assets/17872879/a2380f77-a1aa-4fec-b8f2-6d2147a63a48">

To do the same on F# I must create separate function.
<img width="769" alt="image" src="https://github.com/fsharp/fslang-suggestions/assets/17872879/3310ddd5-602e-4698-9be5-fac7bacccc52">

**Comment by smoothdeveloper** [comment]
@oleksandr-bilyk I think right now it does already do it, if you pass the lambda in the call.

Also, should this work with any delegate type instead? I'm not keen on "Func is better" stuff which is already having few edge cases in overload resolution.

**Comment by smoothdeveloper** [comment]
Likely a special case for #1131

**Comment by brianrourkeboll** [comment]
You can write some code to work around this if you want, like I did in https://github.com/brianrourkeboll/FSharp.AspNetCore.WebAppBuilder; a few others have experimented with similar things, I believe, including @lucasteles, who opened #1131.

**Comment by gurustron** [comment]
Also note that AFAIK the attributes currently are not captured, i.e. since `FromServices` will be correctly inferred everything will work as expected.

**Comment by lucasteles** [comment]
@brianrourkeboll Yes, I ended up creating this https://github.com/lucasteles/FSharp.MinimalApi also
